agent.go

```diff
@@ -381,7 +381,41 @@ func (n *network) leaveCluster() error {
 	return c.agent.networkDB.LeaveNetwork(n.ID())
 }
 
-func (ep *endpoint) addToCluster() error {
+func (ep *endpoint) addDriverInfoToCluster() error {
+	n := ep.getNetwork()
+	if !n.isClusterEligible() {
+		return nil
+	}
+	if ep.joinInfo == nil {
+		return nil
+	}
+
+	for _, te := range ep.joinInfo.driverTableEntries {
+		if err := n.ctrlr.agent.networkDB.CreateEntry(te.tableName, n.ID(), te.key, te.value); err != nil {
```

Comment:
agent can come and go.
Please acquire the agent pointer under lock before getting into the loop.